src/core/mesh/qgsmeshdatagenerator.h

```diff
@@ -0,0 +1,84 @@
+/***************************************************************************
+                         qgsmeshdatagenerator.h
+                         ---------------------
+    begin                : June 2020
+    copyright            : (C) 2020 by Vincent Cloarec
+    email                : vcloarec at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMESHGENERATOR_H
+#define QGSMESHGENERATOR_H
+
+#include <QString>
+#include <QMap>
+
+#include "qgis_core.h"
+
+#define SIP_NO_FILE
+
+class QgsMeshDatasetGroup;
+class QgsMeshLayer;
+
+/**
+ * \ingroup core
+ *
+ * Abstract class that is an interface for mesh data generator.
+ *
+ * Mesh generator can be used to create mesh data as virtual dataset group.
+ *
+ * \since QGIS 3.16
+ */
+class CORE_EXPORT QgsMeshDataGeneratorInterface
```

Comment:
If I understand correctly, we need this QgsMeshDataGeneratorInterface and QgsMeshDataGeneratorRegistry only because the mesh calculator implementation is inside qgis_analysis library?

I am wondering if it is worth adding the extra complexity with these new classes. And for example, any 3rd party apps would have issues using virtual dataset groups generator as it is only registered within QgisApp... What do you think if we would rather move the mesh calculator from qgis_analysis to qgis_core? Would it be a big problem? I guess the only reason the mesh calculator is inside qgis_analysis is because the raster calculator was already there - but even with raster calculator I think sooner or later we will want to have "virtual" raster bands calculated on the fly...